core/src/main/java/com/scalar/db/storage/common/TableMetadataManager.java

```diff
@@ -18,16 +19,19 @@
   private final LoadingCache<TableKey, Optional<TableMetadata>> tableMetadataCache;
 
   public TableMetadataManager(DistributedStorageAdmin admin, DatabaseConfig config) {
+    CacheBuilder<Object, Object> builder = CacheBuilder.newBuilder();
+    long tableMetadataCacheExpirationTimeSecs = config.getTableMetadataCacheExpirationTimeSecs();
+    if (tableMetadataCacheExpirationTimeSecs > 0) {
+      builder.expireAfterWrite(tableMetadataCacheExpirationTimeSecs, TimeUnit.SECONDS);
```

Comment:
I'm not really sure about the use case of specifying the expiration time. Can you clarify?
From users' perspective, it seems like it should be cached forever or it should be replaced right after it needs to 
 be.

Reply:
The use case is when the table metadata change happens. As we don't have `alter table` feature for now, `create table` and `delete table` are only the use cases. Anyway, I think we will need to handle `alter table` cases in the future, so this cache expiration feature will become more important.